src/ducks/apps/index.js

```diff
@@ -389,42 +390,18 @@ export function fetchInstalledApps(client, lang, fetchingRegistry) {
     const { filterAppType } = storeConfig
     try {
       // Start the HTTP requests as soon as possible
-      let fetchingKonnectors = null
-      let fetchingWebApps = null
-      if (
-        !filterAppType || filterAppType === APP_TYPE.KONNECTOR
-      ) {
-        fetchingKonnectors = fetchUserApps(APP_TYPE.KONNECTOR)
-      }
-      if (
-        !filterAppType || filterAppType === APP_TYPE.WEBAPP
-      ) {
-        fetchingWebApps = fetchUserApps(APP_TYPE.WEBAPP)
-      }
+      const toFetch = [APP_TYPE.KONNECTOR, APP_TYPE.WEBAPP].filter(
+        type => !filterAppType || type === filterAppType
+      )
+      const promises = toFetch.map(type => fetchUserApps(client, type))
       await fetchingRegistry
       dispatch({ type: FETCH_APPS })
-      let installedApps = []
-      if (fetchingWebApps) {
-        let installedWebApps = await fetchingWebApps
-        installedWebApps = installedWebApps.filter(
-          app => !config.notDisplayedApps.includes(app.attributes.slug)
-        )
-        installedApps = installedApps.concat(installedWebApps)
-      }
-      if (fetchingKonnectors) {
-        let installedKonnectors = await fetchingKonnectors
-        installedKonnectors = installedKonnectors.filter(
-          app => !config.notDisplayedApps.includes(app.attributes.slug)
-        )
-        installedApps = installedApps.concat(installedKonnectors)
-      }
-      Promise.all(
-        installedApps.map(app => {
-          return getFormattedInstalledApp(app)
-        })
-      ).then(apps => {
-        return dispatch({ type: FETCH_APPS_SUCCESS, apps, lang })
-      })
+      const shouldBeDisplayed = app => !config.notDisplayedApps.includes(app.attributes.slug)
+      const installedApps = flatten(
+        await Promise.all(promises.filter(Boolean))
+      ).filter(shouldAppBeDisplayed)
+      const apps = await Promise.all(installedApps.map(getFormattedInstalledApp))
```

Comment:
Do we need a second `Promise.all` here? Since we await just before, we should get the apps here, shouldn't we?

Reply:
`getFormattedInstalledApp` was async so the Promise.all was necessary.

I checked again, and getFormattedInstalledApp does not need to be async so I removed it and removed the Promise.all ;)